src/pages/settings/Subscription/CardSection/BillingBanner/EarlyDiscountBanner.tsx

```diff
@@ -0,0 +1,105 @@
+import React, {useEffect, useMemo, useState} from 'react';
+import {View} from 'react-native';
+import {useOnyx} from 'react-native-onyx';
+import Button from '@components/Button';
+import * as Illustrations from '@components/Icon/Illustrations';
+import Text from '@components/Text';
+import useLocalize from '@hooks/useLocalize';
+import useResponsiveLayout from '@hooks/useResponsiveLayout';
+import useStyleUtils from '@hooks/useStyleUtils';
+import useTheme from '@hooks/useTheme';
+import useThemeStyles from '@hooks/useThemeStyles';
+import Navigation from '@libs/Navigation/Navigation';
+import {getEarlyDiscountInfo} from '@libs/SubscriptionUtils';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import BillingBanner from './BillingBanner';
+
+type EarlyDiscountBannerProps = {
+    /** Whether the banner is being displayed on the subscription page. */
+    isSubscriptionPage: boolean;
+};
+
+function EarlyDiscountBanner({isSubscriptionPage}: EarlyDiscountBannerProps) {
+    const theme = useTheme();
+    const styles = useThemeStyles();
+    const StyleUtils = useStyleUtils();
+    const {translate} = useLocalize();
+
+    const [firstDayFreeTrial] = useOnyx(ONYXKEYS.NVP_FIRST_DAY_FREE_TRIAL);
+    const [lastDayFreeTrial] = useOnyx(ONYXKEYS.NVP_LAST_DAY_FREE_TRIAL);
+
+    const initialDiscountInfo = getEarlyDiscountInfo();
+    const [discountInfo, setDiscountInfo] = useState(initialDiscountInfo);
+    const [isDismissed, setIsDismissed] = useState(false);
+    const {shouldUseNarrowLayout} = useResponsiveLayout();
+
+    useEffect(() => {
+        const intervalID = setInterval(() => {
```

Comment:
This doesn't look right, can we omit setInterval and call `getEarlyDiscountInfo` only once when component is mounted?